File: 3.time_complexity/lesson_3_tape_equilibrium.py
def solution(A):
    all_sum = 0
    max_value = 0
    current_half_sum = 0
    for i in A:
        all_sum += i
        max_value += abs(i)
    min_value = max_value
    for index in range(len(A)-1):
        current_half_sum += A[index]
        difference = abs(2*current_half_sum - all_sum)
        if(difference < min_value):
            min_value = difference
    return min_value

File: 3.time_complexity/lesson_3_tape_equilibrium.js
function solution(A) {
    let a = A.length
    let result = 100000
    let prefix_sum = new Array(a).fill(0)
    prefix_sum[0] = A[0]
    for(let i = 1; i < a; i++) {
        prefix_sum[i] = prefix_sum[i-1] + A[i]
    }

    for(let i = 0; i < a; i++) {
        if(i != a-1) {
        let abs = Math.abs(prefix_sum[i] - (prefix_sum[a-1]-prefix_sum[i]))
                if (abs < result) {
                    result = abs
                }
        }
    }
    return result
}